client/src/components/AssignmentBoard/VolunteerBox.jsx

```diff
@@ -1,12 +1,36 @@
 import React, { useState } from 'react';
 import { Box } from '@mui/material';
+import {HiOutlineBellAlert} from 'react-icons/hi2';
+import { BellIconStyle } from '../../styles/components/VolunteerModal.style';
 import VolunteerModal from './VolunteerModal/VolunteerModal';
 import { useStyles } from '../../styles/components/Boxes';
 
-const VolunteerBox = ({ volunteer, projects }) => {
+
+const VolunteerBox = ({ volunteer, projects}) => {
     const [modalOpen, setModalOpen] = useState(false);
+    const [showIndicator, setShowIndicator] = useState(false)
+    
+    const checkLocalStorage = () => {
+        const tasksComplete = JSON.parse(localStorage.getItem('tasksComplete'));
+        return {
+            indicatorViewed: tasksComplete.viewed,
+            indicatorNotViewed: tasksComplete.notViewed
+        } 
+    }
+
+    if (checkLocalStorage().indicatorNotViewed.includes(volunteer.id) && !showIndicator) {
+        setShowIndicator(true)
+    }
+    
+    const openModal = () => {
+        if (showIndicator) {
+            let {indicatorViewed, indicatorNotViewed} = checkLocalStorage()
+            localStorage.setItem("tasksComplete", JSON.stringify({viewed: [...indicatorViewed, volunteer.id], notViewed: indicatorNotViewed.filter(id => id !== volunteer.id)}))
```

Comment:
If you take my suggestion to put all of this logic in the parent, this can be replaced with a function that gets passed as a prop.